navapbc/wic-mt-demo-project-eligibility-screener
# Setup precommit hook

## Ticket

https://wicmtdp.atlassian.net/browse/WMDP-34

## Changes
> What was added, updated, or removed in this PR.

- Install husky as a dev dependency
- Configure husky to run linting and tests as a git precommit hook

Extras:
- Have prettier ignore `app/coverage` files
- Rename `ts:check` to `ts-check` to match other script naming conventions

## Context for reviewers
> Testing instructions, background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers. Explain how the changes were verified.

## Testing
> Screenshots, GIF demos, code examples or output to help show the changes working as expected. ProTip: you can drag and drop or paste images into this textbox.

1. Checkout this branch
2. `cd app`
3. `yarn install`
4. `yarn husky-prepare`
5. Make a change
6. Try to commit the change
7. Check that the linting and tests run before the commit is accepted

<img width="451" alt="Screen Shot 2022-07-28 at 12 53 31 AM" src="https://user-images.githubusercontent.com/67701/181452062-abbf20e8-f50a-47e7-ae33-9cb36ec1bcf6.png">
